src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalController.kt

```diff
@@ -122,6 +128,24 @@ class GoalController(
     }
   }
 
+  @GetMapping
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize(HAS_VIEW_AUTHORITY)
+  @Transactional
+  fun getGoals(
+    @PathVariable @Pattern(regexp = PRISON_NUMBER_FORMAT) prisonNumber: String,
+    @RequestParam(required = false) status: GoalStatus?,
+  ): ResponseEntity<Any> {
+    return goalService.getGoals(
+      GetGoalsDto(prisonNumber, status?.let { goalResourceMapper.fromModelToDto(status) }),
+    ).let { result ->
+      when (result) {
```

Comment:
Fair enough, will add some tests. The reason I hadn't was because I had integration tests covering all the same cases and the unit tests are not currently faster than the spring integration tests and that's because it always has to start the spring gubbins and that's because we can't build using intellij and that's because of the code gen and I didn't have the energy to tackle that particular topic just yet.

Reply:
Theres a strong argument for being able to remove some of the integration tests if we write the unit tests to cover the main code paths here 👍  (which will make the integrations quicker)